Entity/ClientManager.php

```diff
@@ -24,31 +24,31 @@ class ClientManager extends BaseClientManager
     /**
      * @var EntityManager
      */
-    protected $em;
+    protected $entityManager;
 
     /**
-     * @var EntityRepository
+     * @var ObjectManager|EntityRepository
```

Comment:
This should be set to `EntityRepository` only.

Reply:
Problem is, `EntityManager::getRepository` is [documented as returning both](https://github.com/doctrine/doctrine2/blob/master/lib/Doctrine/ORM/EntityManager.php#L705).


Asked @ocramius on Twitter, [he thought it's different](https://twitter.com/Ocramius/status/951845101998833664). Changed at https://github.com/doctrine/doctrine2/commit/99db207a9f1214ad483efd81341279a550526358, don't know why.

Reply:
This should be `EntityRepository`. `EntityRepository` implements `ObjectRepository`, so there is no need to change anything here, as the type is compatible.

Reply:
I think the issue is, you're not guaranteed to get an `EntityRepository` from `EntityManager::getRepository` but an `ObjectRepository`.

That's the reason PHPStan is complaining, might be wrong though. @ondrejmirtes, mind confirming/denying this?

Reply:
Still, `ObjectRepository` is not `ObjectManager`. :)

Reply:
:see_no_evil: now I see what you mean. :+1: That's clearly an error, how on earth did it pass the build.

It's supposed to be `ObjectRepository`.

Reply:
Note that `ObjectRepository|EntityRepository` is the same as just saying `ObjectRepository`.

Reply:
@dkarlovi What exactly is PHPStan complaining about?

Reply:
ObjectRepository cannot be assigned to EntityRepository.

Reply:
Yes, makes sense, so `ObjectRepository` should be used everywhere.